### PR TITLE
Add test for convention properties + mapping to `null`

### PR DESCRIPTION
**Pull request by wolfs**
Adds a small test for a behaviour regarding properties, conventions and mapping I wasn't sure about.

### CODE REVIEW

File: subprojects/model-core/src/test/groovy/org/gradle/api/internal/provider/DefaultPropertyTest.groovy [hunk 1]
```diff
@@ -356,4 +356,16 @@ class DefaultPropertyTest extends PropertySpec<String> {
         1 * transformer.transform("abc") >> "cba"
         0 * _
     }
+
+    def "provider from property with convention can be absent"() {
+        def property = propertyWithDefaultValue(String)
+        def conventionValue = "convention"
+        property.convention(conventionValue)
```
Comment (lptr) [inline subprojects/model-core/src/test/groovy/org/gradle/api/internal/provider/DefaultPropertyTest.groovy:363]: Nit: why the need for the local variable here?

**Review by lptr — APPROVED** [review]
LGTM.

File: subprojects/model-core/src/test/groovy/org/gradle/api/internal/provider/DefaultPropertyTest.groovy (hunk 1 above)
Comment (wolfs) [reply subprojects/model-core/src/test/groovy/org/gradle/api/internal/provider/DefaultPropertyTest.groovy]: Inlined.

**Comment by wolfs** [comment]
@bot-gradle test and merge

**Pull request merged**